packages/angular_devkit/core/src/workspace/workspace.ts

```diff
@@ -55,15 +63,69 @@ export class AmbiguousProjectPathException extends BaseException {
   }
 }
 
+async function _findUp(host: virtualFs.Host, names: string[], from: Path): Promise<Path | null> {
+  if (!Array.isArray(names)) {
+    names = [names];
+  }
+
+  do {
+    for (const name of names) {
+      const p = join(from, name);
+      if (await host.exists(p)) {
+        return p;
+      }
+    }
+
+    from = dirname(from);
+  } while (from && from !== dirname(from));
+
+  return null;
+}
+
 export class Workspace {
+  protected static _workspaceFile = [
+    'angular.json',
+    '.angular.json',
+    'workspace.json',
+    '.workspace.json',
+  ];
+
   private readonly _workspaceSchemaPath = normalize(require.resolve('./workspace-schema.json'));
   private _workspaceSchema: JsonObject;
   private _workspace: WorkspaceSchema;
   private _registry: schema.CoreSchemaRegistry;
 
-  constructor(private _root: Path, private _host: virtualFs.Host<{}>) {
-    this._registry = new schema.CoreSchemaRegistry();
-    this._registry.addPostTransform(schema.transforms.addUndefinedDefaults);
+  constructor(
+    private _root: Path,
+    private _host: virtualFs.Host<{}>,
+    registry?: schema.CoreSchemaRegistry,
+  ) {
+    if (registry) {
+      this._registry = registry;
+    } else {
+      this._registry = new schema.CoreSchemaRegistry();
+      this._registry.addPostTransform(schema.transforms.addUndefinedDefaults);
```

Comment:
If `registry` is provided, does it need to already have this transform? 

If so, we should check and error out.

Reply:
No. Honestly `Registry` should always be required. All the places we  build it and yet it's just best to always use the same instance for everything.

Reply:
The problem is that this is part of a public API so we can't break it.

Reply:
It's marked as experimental.